anxcloud/resource_kubernetes_cluster_test.go

```diff
@@ -0,0 +1,139 @@
+package anxcloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/anexia-it/terraform-provider-anxcloud/anxcloud/testutils/environment"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+	kubernetesv1 "go.anx.io/go-anxcloud/pkg/apis/kubernetes/v1"
+	"go.anx.io/go-anxcloud/pkg/utils/pointer"
+)
+
+func TestAccAnxCloudKubernetesCluster(t *testing.T) {
+	//t.Skip("Needs some optimization")
```

Comment:
Do we want the tests to always run? This could be feasible if we have zero-replica node pools.

Reply:
Two test modes:
* on feature branches: integration test not going towards real Engine API, just checking if we send the expected requests and process replies correctly
* on `main`: real e2e test, create a fully working k8s cluster with one node and check if everything is working

Reply:
@LittleFox94 could you please add the [`-short`](https://pkg.go.dev/testing#Short) flag to our branch testing workflow?

Reply:
did that, but now thinking if mock tests should run in `make test` already

or if instead all acceptance tests should test for short mode and either run mock tests or don't do their long running things

Reply:
any input on this would be nice @marioreggiori

Reply:
Is your thought limited to the Kubernetes resources, or in general?

Reply:
The mocked Kubernetes resource test is already covered by running `make test` as it has `IsUnitTest` set to `true`.